.github/workflows/migrate.yml

```diff
@@ -0,0 +1,78 @@
+name: migrate
+
+on:
+  schedule:
+    - cron: "0 7 * * 1"
+  workflow_dispatch:
+
+permissions:
+    contents: write
+    pull-requests: write
+
+jobs:
+    main:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  filter: tree:0
+                  fetch-depth: 0
+            
+            - name: Setup git user
+              shell: bash
+              run: |
+                git config --global user.name 'amidostacks'
+                git config --global user.email 'stacks@amido.com'
+            
+            - uses: pnpm/action-setup@v4
+              with:
+                  version: '10'
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: 'pnpm'
+
+            - run: pnpm install --frozen-lockfile
+
+            - name: Get latest Nx version
+              id: nx-version
+              shell: bash
+              run: |
+                # Store output if NX_VERSION is different from the current version
+                echo "NX_VERSION=$(npm show nx version)" >> $GITHUB_OUTPUT
+                echo "CURRENT_NX_VERSION=$(node -p \"require('./package.json').dependencies.nx\")" >> $GITHUB_OUTPUT
+                echo "SHOULD_MIGRATE=$( [[ $NX_VERSION != $CURRENT_NX_VERSION ]] && echo $? )" >> $GITHUB_OUTPUT
+            
+            - name: Migrate
+              if: steps.nx-version.outputs.should_migrate
```

Comment:
are we 100% sure that this will always yield and output that will be parsed as true/false? Would it make sense to compare `should_migrate` to the expected value `true`?